src/components/ui/members/member-profile-container.tsx

```diff
@@ -0,0 +1,96 @@
+"use client"
+
+import { useState } from "react"
+import { useRouter } from "next/navigation"
+import { useUser, useUserMutation } from "@/lib/api/users"
+import { ProfileInfoUpdateForm } from "@/components/ui/members/profile-info-update-form"
+import { PasswordUpdateForm } from "@/components/ui/members/password-update-form"
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card"
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs"
+import { useAuthStore } from "@/lib/providers/auth-store-provider"
+import { Id } from "@/types/general"
+import { User, NewUserPassword } from "@/types/user"
+import { useUserPasswordMutation } from "@/lib/api/users"
+import { useLogoutUser } from "@/lib/hooks/use-logout-user"
+
+export function MemberProfileContainer({ userId }: { userId: Id }) {
+    const { userId: currentUserId, logout } = useAuthStore((state) => state)
+    const { user, isLoading, refresh } = useUser(userId)
+    const { update: updateUser, isLoading: isUpdating } = useUserMutation()
+    const { update: updatePassword } = useUserPasswordMutation()
+    const [activeTab, setActiveTab] = useState("profile")
+    const logoutUser = useLogoutUser()
+
+    const handleProfileUpdate = async (
+        updatedUser: User
+    ) => {
+        try {
+            await updateUser(userId, updatedUser)
+            refresh()
+        } catch (error) {
+            console.error("Error updating profile:", error)
+        }
+    }
+
+    const handlePasswordUpdate = async (updatedPassword: NewUserPassword) => {
+        try {
+            await updatePassword(userId, updatedPassword)
+            // Logout the user when the password is updated
+            logoutUser()
+        } catch (error) {
+            console.error("Error updating password:", error)
+        }
+    }
```

Comment:
I'd like to suggest using a [Sonner](https://ui.shadcn.com/docs/components/sonner) element from shadcn to display a status message like "Profile updated successfully." or "New password set successfully." after a successful operation for both.

Also use this for the error setting/updating cases as well.

This should be a straightforward addition that will address a real source of confusion.

Reply:
I attempted to add Sonner here and ran into some dependency issues. I'd like to defer this for now if that's ok with you and maybe approach it as it's own issue. There are certainly other places in the UX that would benefit from a wholistic view and design for this I think.

Reply:
@calebbourg That's fine if you're stuggling with Sonner specifically, but we do need feedback that the profile information was updated successfully/unsuccessfully even if it's a black text/red text that displays at the top or bottom of the form.